### ISSUE TITLE
StackOverflow when server responds incorrectly to PASS command

### ISSUE THREAD

**Issue by KoalaBear84** [robinrodricks/FluentFTP#620]
**FTP OS:** Unknown

**FTP Server:** BFTPd

**Computer OS:** Windows 10

**FluentFTP Version:** 32.4.5

Both in .NET Framework and .NET Core/Standard.

Followup to #610 

Found a source which will get in the StackOverflow eventually, and already got a couple of times.

```csharp
Console.Title = "FluentFtp StackOverflowReproduce";
Console.WriteLine("Started!");

FtpTrace.LogToFile = "FluentFTP.txt";

FtpTrace.LogUserName = false;   // Hide FTP user names
FtpTrace.LogPassword = false;   // Hide FTP passwords
FtpTrace.LogIP = false; 	// Hide FTP server IP addresses

Console.WriteLine("Creating FtpClient...");
FtpClient ftpClient = new FtpClient(new Uri("ftp://103.100.31.103:21"));
Console.WriteLine("Created FtpClient");

Console.WriteLine("Connecting...");
ftpClient.Connect();
Console.WriteLine("Connected");

Console.WriteLine("Finished!");
Console.ReadKey(intercept: true);
```

**Logs :**

```
# Connect()
Status:   Connecting to ***:21
Response: 220 bftpd 2.2.1 at 192.168.0.1 ready.
Status:   Detected FTP server: BFTPd
Command:  USER ***
Response: 331 Password please.
Command:  PASS ***
Response: 331 Password please.
Status:   There is stale data on the socket, maybe our connection timed out or you did not call GetReply(). Re-connecting...
Status:   The stale data was: 530 Login incorrect.
Status:   Disposing FtpSocketStream...

# Connect()
Status:   Connecting to ***:21
Response: 220 bftpd 2.2.1 at 192.168.0.1 ready.
Status:   Detected FTP server: BFTPd
Command:  USER ***
Response: 331 Password please.
Command:  PASS ***

# Connect()
Status:   Connecting to ***:21
Response: 220 bftpd 2.2.1 at 192.168.0.1 ready.
Status:   Detected FTP server: BFTPd
Command:  USER ***
Response: 331 Password please.
Command:  PASS ***
Response: 331 Password please.
Status:   There is stale data on the socket, maybe our connection timed out or you did not call GetReply(). Re-connecting...
Status:   The stale data was: 530 Login incorrect.
Status:   Disposing FtpSocketStream...

# Connect()
Status:   Connecting to ***:21
Response: 220 bftpd 2.2.1 at 192.168.0.1 ready.
Status:   Detected FTP server: BFTPd
Command:  USER ***
Response: 331 Password please.
Command:  PASS ***
Response: 331 Password please.
Status:   There is stale data on the socket, maybe our connection timed out or you did not call GetReply(). Re-connecting...
Status:   The stale data was: 530 Login incorrect.
Status:   Disposing FtpSocketStream...

# Connect()
Status:   Connecting to ***:21
Response: 220 bftpd 2.2.1 at 192.168.0.1 ready.
Status:   Detected FTP server: BFTPd
Command:  USER ***
Response: 331 Password please.
Command:  PASS ***
Response: 331 Password please.
Status:   There is stale data on the socket, maybe our connection timed out or you did not call GetReply(). Re-connecting...
Status:   The stale data was: 530 Login incorrect.
Status:   Disposing FtpSocketStream...

# Connect()
Status:   Connecting to ***:21
Response: 220 bftpd 2.2.1 at 192.168.0.1 ready.
Status:   Detected FTP server: BFTPd
Command:  USER ***
Response: 331 Password please.
Command:  PASS ***
Response: 331 Password please.
Status:   There is stale data on the socket, maybe our connection timed out or you did not call GetReply(). Re-connecting...
Status:   The stale data was: 530 Login incorrect.
Status:   Disposing FtpSocketStream...

# Connect()
Status:   Connecting to ***:21
Response: 220 bftpd 2.2.1 at 192.168.0.1 ready.
Status:   Detected FTP server: BFTPd
Command:  USER ***
Response: 331 Password please.
Command:  PASS ***
Response: 331 Password please.
Status:   There is stale data on the socket, maybe our connection timed out or you did not call GetReply(). Re-connecting...
Status:   The stale data was: 530 Login incorrect.
Status:   Disposing FtpSocketStream...
```

I will continue to keep sort of 'reconnecting' indefinitely.

Screen recording:
![screenrecording](https://user-images.githubusercontent.com/5886900/93742826-19fd2f00-fbef-11ea-91e9-00ca648c05f1.gif)

Reproduce project:
[ConsoleApp_20200921_0929_FluentFtpStackOverflowReproduce.zip](https://github.com/robinrodricks/FluentFTP/files/5253972/ConsoleApp_20200921_0929_FluentFtpStackOverflowReproduce.zip)

**Comment by robinrodricks** [comment]
It seems your server is malfunctioning. See:

```
Command:  PASS ***
Response: 331 Password please.
```
Why is it sending 331 after password has already been sent?

**Comment by KoalaBear84** [comment]
This is also not my FTP server.

I have no idea, can it be solved somehow? (By ignoring it?)

**Comment by robinrodricks** [comment]
Does this work?

[FluentFTP.32.4.6_BETA2.zip](https://github.com/robinrodricks/FluentFTP/files/5333280/FluentFTP.32.4.6_BETA2.zip)

**Comment by KoalaBear84** [comment]
Thanks, yes, this works, no more endless connecting.

**Comment by robinrodricks** [comment]
Released in [32.4.6](https://www.nuget.org/packages/FluentFTP/32.4.6)
